contrib/babelfishpg_tsql/runtime/functions.c

```diff
@@ -2423,3 +2426,59 @@ pg_extension_config_remove(PG_FUNCTION_ARGS)
 
 	PG_RETURN_VOID();
 }
+
+/*
+ * The EOMONTH function is a Transact-SQL function in SQL Server that returns 
+ * the last day of the month of a specified date, with an optional offset.
+ */
+Datum
+EOMONTH(PG_FUNCTION_ARGS)
+{
+    int year, month, day;
+    DateADT date = PG_GETARG_DATEADT(0);
+    int offset = PG_GETARG_INT32(1);
+
+    /* Convert the date to year, month, day */
+    j2date(date + POSTGRES_EPOCH_JDATE, &year, &month, &day);
+
+    /* Adjust the month based on the offset */
+    month += offset;
+    // Check if the new month is greater than 0.
```

Comment:
Please modify the comment style here

Reply:
Yes, thanks for pointing that. Will do that in the next revision.